### PR TITLE
Adds holo pipes

### PR DESCRIPTION
**Pull request by Arthurdentist**
Credit to @RealestEstate for sprites, still need to make them acquirable from somewhere but I will do it later 

:cl: 
 * rscadd: Added holo-pipes as a smoke free alternative to smoking. They are available as premium content from cigarette machines.

### CODE REVIEW

**Review by MadmanMartian — DISMISSED** [review]
fix the partial pathing.

**Comment by MadmanMartian** [comment]
the on and off sprites look the same in the icondiffbot

**Comment by RealestEstate** [comment]
@MadmanMartian it's really hard to tell on a white background, but the on-holopipe has some holographic smoke coming off it.

**Comment by Kurfursten** [comment]
This is... basically a vape, isn't it?

**Comment by Arthurdentist** [comment]
@Kurfursten it isn't a vape either

**Comment by Arthurdentist** [comment]
this PR is finished, can buy holo pipes as premium content from the cig machines

File: code/modules/clothing/masks/miscellaneous.dm [hunk 1]
```diff
@@ -225,3 +225,35 @@ obj/item/clothing/mask/joy
...
+
+
+/obj/item/clothing/mask/holopipe/proc/activate(var/mob/user as mob)
+	if(!user.incapacitated())
+		src.activated = !src.activated
+		if(src.activated)
+			icon_state = "holopipe_on"
+			item_state = "holopipe_on"
+			to_chat(user, "You activate the holo pipe.")
+			user.regenerate_icons() //update sprites if toggled whilst the holo pipe is being worn
```
Comment (DamianX) [inline code/modules/clothing/masks/miscellaneous.dm]: `user.update_inv_wear_mask()` should be sufficient

File: code/modules/clothing/masks/miscellaneous.dm [hunk 2]
```diff
@@ -225,3 +225,35 @@ obj/item/clothing/mask/joy
...
+		else
+			icon_state = "holopipe_off"
+			item_state = "holopipe_off"
+			to_chat(user, "You deactivate the holo pipe.")
+			user.regenerate_icons()
```
Comment (DamianX) [inline code/modules/clothing/masks/miscellaneous.dm]: `user.update_inv_wear_mask()` should be sufficient

File: code/modules/clothing/masks/miscellaneous.dm [hunk 3]
```diff
@@ -225,3 +225,35 @@ obj/item/clothing/mask/joy
...
+
+/obj/item/clothing/mask/holopipe/attack_self(var/mob/user as mob)
```
Comment (DamianX) [inline code/modules/clothing/masks/miscellaneous.dm]: remove the `as mob` part, it does nothing

File: code/modules/clothing/masks/miscellaneous.dm [hunk 4]
```diff
@@ -225,3 +225,35 @@ obj/item/clothing/mask/joy
...
+	activate(user)
+
+/obj/item/clothing/mask/holopipe/verb/activate_pipe()
+	set category = "Object"
+	set name = "Toggle pipe"
+	set src in usr
+	activate(usr)
```
Comment (DamianX) [inline code/modules/clothing/masks/miscellaneous.dm]: I think you need to check for incapacitation before calling this

**Review by DamianX — APPROVED** [review]

File: code/modules/clothing/masks/miscellaneous.dm [hunk 5]
```diff
@@ -225,3 +225,36 @@ obj/item/clothing/mask/joy
...
+			user.update_inv_wear_mask()
+
+/obj/item/clothing/mask/holopipe/attack_self(var/mob/user)
+	activate(user)
+
+/obj/item/clothing/mask/holopipe/verb/activate_pipe()
+	set category = "Object"
+	set name = "Toggle pipe"
+	set src in usr
+	if(!usr.incapacitated())
```
Comment (DamianX) [inline code/modules/clothing/masks/miscellaneous.dm:259]: ah, you were already checking inside activate(), my bad

**Pull request merged**